examples/README.md

```diff
@@ -0,0 +1,141 @@
+in the headers.
+
+* Schema validation: after the contents of the message has been extracted, it is validated using the 
+schemas provided by the Redpanda Schema registry. To do that, it reads from the headers ... the subject name
+and version number of the schema the message was encoded into. With these two fields it will request
+the Redpanda service for the schema published using that subject and version number. The schema downloaded
+is then processed by the Fastavro library to validate the correction of the message contents.
+
+* Message processing: if the validation is correct, a new instance of the Processor class defined in 
+the PROCESSORS field of the config is created and the library will call the method process() with the
+contents of the message. Any required behaviour for the processing of the message can be added inside this
+method. If the process is correct (returns True), the message is purged, but if the process is incorrect (returns False) the message is sent to the dead letters queue.
+
+## Setting up a consumer
+
+To set up a consumer first we need to define the configuration settings that the consumer will use to run.
+These are the settings required for a consumer:
+
+You can copy file ```examples/example1/config.py``` and modify with your settings.
+
+Second, we have to define a new processor for the messages to read. To do it, you have add a new entry in the
+```examples/example1/config.py``` we created before with the name of the class we want to use. For example:
+
+```python
+# Hash that maps each subject name with a processor class that will be instantiated when
+# we consume a message using that subject name (specified in header from rabbitmq: 'subject')
+PROCESSORS={
+    'example_1_message': Example1MessageProcessor 
+}
+```
+
+This will define a new processor class Example1MessageProcessor that will process any new messages where the subject name is 'example_1_message'.
+
+In ```examples/example1/processors.py``` we have an example of a processor class skeleton you can use to implement new 
+process for messages received with subject 'example_1_message':
+
+```python
+class Example1MessageProcessor:
+    def __init__(self, schema_registry, basic_publisher, config):
+        return
+
+    def process(self, message):
+        print(message.message)
+        return True
+```
+
+The method ```process``` receives a message as argument after being unserialized, so you can perform there the process you require for the message.
+
+Third we have to start our consumer in the app we are working on.  A LabShare consumer will run as a separate thread from the main application after starting. As such there are different options to control how the consumer is performing where. the most basic could be to check inside an infinite loop. For example:
+
+```python
+import time
+from lab_share_lib.rabbit.rabbit_stack import RabbitStack
+
+settings = "config"
+
+if __name__ == "__main__":
+    print("Starting TOL consumer")
```

Comment:
Not sure if you want to include `TOL` in this library?

Reply:
I didnt see, thanks